internal/maintenance/reindexer.go

```diff
@@ -105,31 +105,48 @@ func (s *Reindexer) Start(ctx context.Context) error {
 		s.Logger.DebugContext(ctx, s.Name+logPrefixRunLoopStarted)
 		defer s.Logger.DebugContext(ctx, s.Name+logPrefixRunLoopStopped)
 
-		// On each run, we calculate the new schedule based on the previous run's
-		// start time. This ensures that we don't accidentally skip a run as time
-		// elapses during the run.
```

Comment:
Moved this comment down below because I think it's more salient down there where the second+ next run ats are being calculated.